File: 0061-rotate-list/0061-rotate-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode rotateRight(ListNode head, int k) {
        if(head==null || k==0 || head.next==null)
            return head;
        int count=0;
        ListNode temp=head;
        while(temp!=null){
            temp=temp.next;
            count++;
        }
        k%=count;
        if(k==0)
            return head;
        ListNode sec=null;
        temp=head;
        for(int i=1;i<count-k;i++){
            temp=temp.next;
        }
        sec=temp.next;
        temp.next=null;
        temp=sec;
        while(temp.next!=null)
            temp=temp.next;
        temp.next=head;
        return sec;
    }
}

File: 0061-rotate-list/0061-rotate-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* rotateRight(ListNode* head, int k) {
        ListNode *temp=head;
        int N=0;
        while(temp!=NULL){
            N++;
            temp=temp->next;
        }
        if(N==0)
            return NULL;
        ListNode* sec=NULL;
        temp=head;
        k%=N;
        for(int i=1;i<N-k;i++){
            temp=temp->next;
        }
        sec=temp->next;
        temp->next=NULL;
        temp=sec;
        while(temp!=nullptr && temp->next!=NULL){
            temp=temp->next;
        }
        if(temp!=nullptr){
            temp->next=head;
        }else{
            return head;
        }
        return sec;
    }
};